src/main/java/core/basesyntax/supplier/FigureSupplier.java

```diff
@@ -0,0 +1,52 @@
+package core.basesyntax.supplier;
+
+import core.basesyntax.Colors;
+import core.basesyntax.Figure;
+import core.basesyntax.Shapes;
+import core.basesyntax.figure.Circle;
+import core.basesyntax.figure.IsoscelesTrapezoid;
+import core.basesyntax.figure.Rectangle;
+import core.basesyntax.figure.RightTriangle;
+import core.basesyntax.figure.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MIN_RANDOM_UNITS = 1;
+    private static final int MAX_RANDOM_UNITS = 11;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        String randomShape = getRandomShape();
+        String randomColor = colorSupplier.getRandomColor();
+        int randomUnits = random.nextInt(MIN_RANDOM_UNITS, MAX_RANDOM_UNITS);
+
+        switch (randomShape) {
+            case "SQUARE":
+                return new Square(randomShape, randomColor, randomUnits);
```

Comment:
It is not necessary to write return in all cases, it is possible to create a variable on method-level and populate it in cases